docs/build-javascaladocs.sh

```diff
@@ -1,23 +1,58 @@
-#! /bin/bash
-
-VERSION=$1
-
-echo $VERSION
-
-if [[ -z "${VERSION}" ]]; then
-  echo "Version not defined"
-else
-  (cd target && \
-    rm -rf cloudflow && \
-    git clone --depth=100 --branch "$VERSION" https://github.com/lightbend/cloudflow.git && \
-    cd cloudflow && \
-    export DIR=$(yq e '.version' docs/docs-source/docs/antora.yml) && \
-    echo $DIR && \
-    cd core && \
-    (sbt -mem 2048 clean unidoc || true) && \
-    cd ../../../ && \
-    ls -al "./target/cloudflow/core/target" && \
-    ls -al "./target/cloudflow/core/target/scala-2.12" && \
-    mv "./target/cloudflow/core/target/scala-2.12/unidoc" "./target/staging/docs/$DIR/api/scaladoc" && \
-    mv "./target/cloudflow/core/target/javaunidoc" "./target/staging/docs/$DIR/api/javadoc")
+#!/usr/bin/env bash
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+ROOT_DIR=$( cd "${SCRIPT_DIR}/.." && pwd )
+DOCS_DIR="$ROOT_DIR/docs"
+
+# Keep this consistent with the `work_dir` in the Makefile.
+TARGET_DIR="$DOCS_DIR/target"
+CLOUDFLOW_CLONE_DIR="${TARGET_DIR}/cloudflow"
+
+echo "SCRIPT_DIR          => $SCRIPT_DIR"
+echo "ROOT_DIR            => $ROOT_DIR"
+echo "DOCS_DIR            => $DOCS_DIR"
+echo "TARGET_DIR          => $TARGET_DIR"
+echo "CLOUDFLOW_CLONE_DIR => $CLOUDFLOW_CLONE_DIR"
+
+versions=$(yq e '.content.sources[0].branches[]' "$DOCS_DIR/docs-source/site.yml")
+
+if [ -z "$versions" ]; then
+  echo "No version branches found in $DOCS_DIR/docs-source/site.yml. See file content below:"
+  cat "$DOCS_DIR/docs-source/site.yml"
+  exit 1
+fi
+
+# Recreating the target dir before generating the new docs.
+rm -rf "$TARGET_DIR/cloudflow"
+mkdir -p -v "$TARGET_DIR"
+
+git clone https://github.com/lightbend/cloudflow.git "$CLOUDFLOW_CLONE_DIR"
+
+FAILURES=()
+
+for version in $versions; do
+  echo "Building docs for version $version"
+  cd "$CLOUDFLOW_CLONE_DIR" || exit 1 # Maybe if failed to clone the cloudflow repo.
+  git checkout --quiet "$version"
```

Comment:
Changing the version using a simple `checkout` would be nice, have you checked that there are no residual files in the target directories? I recall having troubles with cleanups ...

Reply:
"should" ok, let's give this approach a go.